unit-tests/post-processing/test-rotation-filter.py

```diff
@@ -0,0 +1,133 @@
+#test:donotrun:dds
+#test:donotrun:!nightly
+
+from rspy import test, repo
+import pyrealsense2 as rs
+import numpy as np
+
+# Parameters for frame creation and depth intrinsics
+input_res_x = 640
+input_res_y = 480
+focal_length = 600
+depth_units = 0.001
+stereo_baseline_mm = 50
+frames = 5  # Number of frames to process
+
+
+# Function to create depth intrinsics directly
+def create_depth_intrinsics():
+    depth_intrinsics = rs.intrinsics()
+    depth_intrinsics.width = input_res_x
+    depth_intrinsics.height = input_res_y
+    depth_intrinsics.ppx = input_res_x / 2.0
+    depth_intrinsics.ppy = input_res_y / 2.0
+    depth_intrinsics.fx = focal_length
+    depth_intrinsics.fy = focal_length
+    depth_intrinsics.model = rs.distortion.brown_conrady
+    depth_intrinsics.coeffs = [0, 0, 0, 0, 0]
+    return depth_intrinsics
+
+
+# Function to create a video stream with specified parameters
+def create_video_stream(depth_intrinsics):
+    vs = rs.video_stream()
+    vs.type = rs.stream.depth
+    vs.index = 0
+    vs.uid = 0
+    vs.width = input_res_x
+    vs.height = input_res_y
+    vs.fps = 30
+    vs.bpp = 2
+    vs.fmt = rs.format.z16
+    vs.intrinsics = depth_intrinsics
+    return vs
+
+
+# Function to create a synthetic frame
+def create_frame(depth_stream_profile, index):
+    frame = rs.software_video_frame()
+    data = np.arange(input_res_x * input_res_y, dtype=np.uint16).reshape((input_res_y, input_res_x))
+    frame.pixels = data.tobytes()
+    frame.bpp = 2
+    frame.stride = input_res_x * 2
+    frame.timestamp = index * 33
+    frame.domain = rs.timestamp_domain.system_time
+    frame.frame_number = index
+    frame.profile = depth_stream_profile.as_video_stream_profile()
+    frame.depth_units = depth_units
+    return frame
+
+
+# Function to validate rotated results based on the angle
+def validate_rotation_results(filtered_frame, angle):
+    rotated_height = filtered_frame.profile.as_video_stream_profile().height()
+    rotated_width = filtered_frame.profile.as_video_stream_profile().width()
+
+    # Reshape the rotated data according to its actual dimensions
+    rotated_data = np.frombuffer(filtered_frame.get_data(), dtype=np.uint16).reshape((rotated_height, rotated_width))
+
+    # Original data for comparison
+    original_data = np.arange(input_res_x * input_res_y, dtype=np.uint16).reshape((input_res_y, input_res_x))
+
+    # Determine the expected rotated result based on the angle
+    if angle == 90:
+        expected_data = np.rot90(original_data, k=-1)  # Clockwise
+    elif angle == 180:
+        expected_data = np.rot90(original_data, k=2)  # 180 degrees
+    elif angle == -90:
+        expected_data = np.rot90(original_data, k=1)  # Counterclockwise
+
+    # Convert numpy arrays to lists before comparison
+    rotated_list = rotated_data.flatten().tolist()
+    expected_list = expected_data.flatten().tolist()
+
+    # Compare the flattened lists
+    test.check_equal_lists(rotated_list, expected_list)
```

Comment:
The data is all 0?

Reply:
Maybe I am mistaken but if this is the case,
You take 640x480 of 0's and flatten it
And take 480x460 and flatten it,
You will get the test pass no matter each angle you rotate no?

Reply:
No, the data ranging from 0 to 307,199 (calculated as input_res_x * input_res_y), reshaped to match the resolution (input_res_y, input_res_x).  looks like:
array([[     0,      1,      2, ...,    637,    638,    639],
       [   640,    641,    642, ...,   1277,   1278,   1279],
       ...,
       [306560, 306561, 306562, ..., 307197, 307198, 307199]], dtype=uint16)